inc/gantt/datafactory.class.php

```diff
@@ -0,0 +1,198 @@
+    *
+    * @param array @itemArray Array holding the task items
+    * @param integer @projectId ID of the project
+    */
+   function getProjectTasks(&$itemArray, $projectId) {
+      $taskRecords[] = \ProjectTask::getAllForProject($projectId);
+      foreach ($taskRecords[0] as $record) {
+         $task = new \ProjectTask();
+         $task->getFromDB($record["id"]);
+         if (!$task->canViewItem() || $record['is_template'] == 1) {
+            continue;
+         }
+         array_push($itemArray, $this->populateGanttItem($record, "task"));
+      }
+   }
+
+   /**
+    * Function used to get all subtasks of a task
+    *
+    * @param array @itemArray Array holding the task items
+    * @param integer @taskId ID of the parent task
+    */
+   function getSubtasks(&$itemArray, $taskId) {
+      $taskRecords[] = \ProjectTask::getAllForProjectTask($taskId);
+      foreach ($taskRecords[0] as $record) {
+         $this->getSubtasks($itemArray, $record["id"]);
+
+         $task = new \ProjectTask();
+         $task->getFromDB($record["id"]);
+         if (!$task->canViewItem() || $record['is_template'] == 1) {
+            continue;
+         }
+         array_push($itemArray, $this->populateGanttItem($record, "task"));
+      }
+   }
+
+   /**
+    * Function used to populate gantt Item objects with projects/tasks/milestones data
+    *
+    * @param $record Project or task record from database
+    * @param string $type Specifies the type of the record (project, task or milestone)
+    *
+    * @return Item instance
+    */
+   function populateGanttItem($record, $type) {
+      if (isset($record['is_milestone']) && $record['is_milestone'] > 0) {
+         $type = 'milestone';
+      }
+
+      $parentTaskUid = "";
+      if (($type == 'task' || $type == 'milestone') && $record["projecttasks_id"] > 0) {
+         $parentTask = new \ProjectTask();
+         $parentTask->getFromDB($record["projecttasks_id"]);
+         $parentTaskUid = $parentTask->fields["uuid"];
+      }
+
+      $item = new Item();
+      $item->id = ($type == "project" || $type == "root-project") ? $record['id'] : $record['uuid'];
+      $item->type = ($type == "root-project") ? "project" : $type;
+      $item->parent = ($type == "root-project") ? 0 : (($type == "project") ? $record['projects_id'] : ($record["projecttasks_id"] > 0 ? $parentTaskUid : $record['projects_id']));
+      $item->linksource_id = ($item->type != "project") ? $record["projecttasks_id"] : 0;
+      $item->linktask_id = ($item->type != "project") ? $record["id"] : 0; // parent task id to search for by child->linksource_id
+      $item->start_date = $record['plan_start_date'] ?? $_SESSION['glpi_currenttime'];
+      $item->end_date = $record['plan_end_date'] ?? date('Y-m-d H:i:s', strtotime($item->start_date . ' + 1 day'));
+      $item->text = $record['name'];
+      $item->content = isset($record['content']) ? html_entity_decode($record['content']) : "";
```

Comment:
To get safe HTML, you have to use.
```suggestion
      $item->content = isset($record['content']) ? \Html::clean($record['content'], false, 1) : "";
```